k2/csrc/context.h

```diff
@@ -550,15 +550,51 @@ class ParallelRunner {
   // will call CudaStreamOverride::OverrideStream() and replace it
   // with this stream automatically.
   cudaStream_t NewStream() {
-    // TODO
-    return kCudaStreamInvalid;
+    DeviceType d = c_->GetDeviceType();
+    if (d == kCpu) {
+      return kCudaStreamInvalid;
+    } else {
+      K2_CHECK_EQ(d, kCuda);
+      // create event
+      cudaEvent_t event;
+      auto ret = cudaEventCreate(&event);
+      K2_CHECK_CUDA_ERROR(ret);
+      events_.push_back(event);
```

Comment:
Mm, this event stuff is not quite right.
Firstly, you can create an even in c_->GetCudaStream() when this object is created, and have all streams that you create wait on it as soon as they are created. 
Next, you only need to create events in the spawned streams in the destructor of ParallelRunner.  The idea is to force c_->GetCudaStream() to wait on the jobs done in the spawned streams, so the event must be created after those jobs have been submitted.

Reply:
.. so you don't need an events_ member, but you can have an event_ member containing just one event.

Reply:
Why do we need to wait event on `c_->GetCudaStream` in created streams when the object is created? Can't we suppose all jobs on `c_->GetCudaStream` have been done before creating these streams?

Reply:
Updated, please review again, thanks